pkg/asset/installconfig/aws/validation.go

```diff
@@ -330,15 +330,50 @@ var requiredServices = []string{
 // ValidateForProvisioning validates if the install config is valid for provisioning the cluster.
 func ValidateForProvisioning(session *session.Session, ic *types.InstallConfig, metadata *Metadata) error {
 	allErrs := field.ErrorList{}
-	allErrs = append(allErrs, validateExistingHostedZone(session, ic, metadata)...)
+	if ic.AWS.HostedZone == "" {
+		allErrs = append(allErrs, validatePreExistingPublicDNS(session, ic, metadata)...)
+	} else {
+		allErrs = append(allErrs, validateExistingHostedZone(session, ic, metadata)...)
+	}
 	return allErrs.ToAggregate()
 }
 
-func validateExistingHostedZone(session *session.Session, ic *types.InstallConfig, metadata *Metadata) field.ErrorList {
-	if ic.AWS.HostedZone == "" {
+// validate no pre-existing DNS records exist in the public DNS zone
+func validatePreExistingPublicDNS(session *session.Session, ic *types.InstallConfig, metadata *Metadata) field.ErrorList {
+	if ic.Publish == types.InternalPublishingStrategy {
 		return nil
 	}
 
+	baseDomainPath := field.NewPath("aws", "baseDomain")
+	baseDomainZone, err := GetPublicZone(session, ic.BaseDomain)
+	if err != nil {
+		return field.ErrorList{
+			field.Invalid(baseDomainPath, ic.BaseDomain, "cannot find base domain"),
+		}
+	}
+
+	allErrs := field.ErrorList{}
+	dottedClusterDomain := ic.ClusterDomain() + "."
```

Comment:
We should leave the details of whether the cluster domain should be dotted to the inners of the `getProblematicRecords` and `isHostedZoneDomainParentOfClusterDomain` functions. In other words, change those two functions to take in the canonical cluster domain rather than the dotted cluster domain, and have the functions add the dot as appropriate.